src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -693,15 +693,28 @@ private void buildMissingParticipantPanelsForTeam(
 
     }
 
-    private void buildSectionPanelsForForAjaxLoading(List<String> sections) {
+    private void buildSectionPanelsForForAjaxLoading(List<String> sections, InstructorFeedbackResultsPageViewType viewType) {
         sectionPanels = new LinkedHashMap<>();
 
-        InstructorFeedbackResultsSectionPanel sectionPanel = new InstructorFeedbackResultsSectionPanel(
-                Const.DEFAULT_SECTION, Const.NO_SPECIFIC_SECTION, true);
-        sectionPanels.put(Const.DEFAULT_SECTION, sectionPanel);
+        boolean hasGeneralFeedback = false;
+        boolean hasTeamsInNoSection = bundle.getTeamsInSectionFromRoster(Const.DEFAULT_SECTION).size() > 0;
+
+        if ((viewType == InstructorFeedbackResultsPageViewType.RECIPIENT_GIVER_QUESTION
+                || viewType == InstructorFeedbackResultsPageViewType.RECIPIENT_QUESTION_GIVER)
+                && bundle.containsResponsesToInstructorOrGeneral()) {
```

Comment:
The if statement tells me that if the viewType is blablabla and the responses contain the feedback to instructors -> then it `hasGeneralFeedback`. A bit confusing.

I think your if statement should be separated into three condition with disjunction.
either feedbackToInstructor || generalFeedback || studentInNoSection.